### PR TITLE
Shorten: Add handler regex to check for valid url

### LINKED ISSUE #2789: Shorten: Triggers on just "shorten" -- Need to check remainder length
This IA is throwing some errors for queries that shouldn't be triggering or showing a result.

The query "shorten" triggers the IA and produces an error: "Use of uninitialized value $q in pattern match (m//) at lib/DDG/Spice/Shorten.pm line 17."

The same happens for "shorten url" and "shorten url for". We need to add more checks to the `handle` statement because it's assuming we have a valid query and defined remainder.

---

IA Page:  http://duck.co/ia/view/shorten 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @danjarvis

### PR DESCRIPTION
**Pull request by NaveenKarippai**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.
- [ ] New Instant Answer: **`New {IA Name} Instant Answer`**
- [x] Improvement
  - [x] Bug fix: **`shorten: Fix #2789 (trigger)`**
  - [ ] Enhancement: **`{IA Name}: {Description of Improvements}`**
- [ ] Non–Instant Answer
  - [ ] Other (Role, Template, Test, Documentation, etc.): **`{SpiceRole/Templates/Tests/Docs}: {Short Description}`**
###### Description of changes

The handler function  of `shorten` spice should check for valid "url"
query input. This would prevent unnecessary triggering of spice for
empty query and invalid url.

Provide an overview of the changes this pull request introduces.
- added regex check for valid "url" on handler function
###### Which issues (if any) does this fix?

See also: #2789

Fixes #NNNN - how specifically does it fix the issue?
###### People to notify (@GuiltyDolphin , @moollaza  )

---

Instant Answer Page: https://duck.co/ia/view/shorten
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @danjarvis

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/DDG/Spice/Shorten.pm [hunk 1]
```diff
@@ -11,6 +11,7 @@ triggers any => 'shorten', 'shorten url', 'short url', 'url shorten';
 my %connector_words = map { $_ => 1 } qw(for of);
 
 handle remainder => sub {
+    return unless ($_ =~ /^((https?|ftp):\/\/)|(w{3}\.)[^\s\/$.?#].[^\s]*$/i);
```
Comment (GuiltyDolphin) [inline lib/DDG/Spice/Shorten.pm]: Lets use `Regexp::Common` with `URI`.

**Comment by moollaza** [comment]
@NaveenKarippai Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

**Comment by moollaza** [comment]
This PR is being closed due to inactivity. If you're still interested in finishing, please let me know and I'll gladly re-open it for you!

**Pull request closed without merging**